### PR TITLE
Link cmsRunJEProf with jemalloc-prof version of jemalloc lib

### PR DESCRIPTION
**Pull request by gartung**
Link `cmsRunJEProf` against `jemalloc-prof` a version of the jemalloc library with profiling enabled during compilation.

### CODE REVIEW

**Comment by gartung** [comment]
@smuzaffar Is this the correct way to link against the profiling enabled jemalloc library?

**Comment by smuzaffar** [comment]
@gartung , I am afraid this is not going to work. Currently we can not have both `jemalloc` and `jemalloc-prof` tools setup at the same time as both provides the same libs (`libjemalloc.so  libjemalloc.so.2`). We need to udpate https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_13_1_X/master/jemalloc-prof.spec so that it renames the libs to e.g. `libjemalloc-prof.so  libjemalloc-prof.so.2` and then we can use `jemalloc-prof` in cmssw

**Comment by gartung** [comment]
pleaee test

File: FWCore/Framework/bin/BuildFile.xml [hunk 1]
```diff
@@ -59,7 +59,7 @@
   <use name="tbb"/>
   <use name="boost"/>
   <use name="boost_program_options"/>
-  <use name="jemalloc"/>
+  <use name="jemalloc-prof"/>
```
Comment (makortel) [inline FWCore/Framework/bin/BuildFile.xml:62]: I wonder if we should make a new executable (e.g. `cmsRunJEProf`)?

Comment (gartung) [reply FWCore/Framework/bin/BuildFile.xml:62]: OK. `cmsRunJE` is there for historical reasons.

**Comment by gartung** [comment]
abort

**Comment by gartung** [comment]
merged into https://github.com/cms-sw/cmssw/pull/40899

**Pull request closed without merging**